sui/manage-coins.js

```diff
@@ -0,0 +1,218 @@
+
+        tx.mergeCoins(firstCoin, remainingCoins);
+        console.log(`Merge ${coins.data.length} coins of type '${chalk.green(coins.data[0].coinType)}'`);
+    }
+
+    static async processCommand(config, chain, options) {
+        const [keypair, client] = getWallet(chain, options);
+        printInfo('Wallet Address', keypair.toSuiAddress());
+
+        const coinTypeToCoins = await CoinManager.getAllCoins(client, keypair.toSuiAddress());
+        CoinManager.printAllCoins(coinTypeToCoins);
+
+        const tx = new TransactionBlock();
+
+        if (options.merge) {
+            await CoinManager.mergeCoin(tx, coinTypeToCoins, options);
+        } else if (options.split) {
+            await CoinManager.splitCoins(tx, coinTypeToCoins, options);
+        }
+
+        const requireBroadcast = options.merge || options.split;
+
+        if (requireBroadcast) {
+            await client.signAndExecuteTransactionBlock({
+                transactionBlock: tx,
+                signer: keypair,
+                options: {
+                    showEffects: true,
+                    showObjectChanges: true,
+                    showContent: true,
+                },
+            });
+
+            console.log(`\nDone`);
+        }
+    }
+
+    static printAllCoins(coinTypeToCoins) {
+        console.log('==== Coins Info ====');
+
+        for (const coinType in coinTypeToCoins) {
+            const coins = coinTypeToCoins[coinType];
+            console.log(`Coin Type: ${chalk.green(coinType)}`);
+            console.log(`Total Balance: ${chalk.green(coins.totalBalance)}`);
+            console.log(`Total Objects: ${chalk.green(coins.data.length)}`);
+        }
+    }
+
+    static checkCoinType(coinType, coinTypeToCoins) {
+        if (coinType && !coinTypeToCoins[coinType]) {
+            console.error(`No coins found for coin type ${coinType}`);
+            process.exit(0);
+        }
+    }
+
+    static isGasToken(coin) {
+        return coin.coinType === '0x2::sui::SUI';
+    }
+
+    static checkSplitAmount(options) {
+        try {
+            parseInt(options.split);
+        } catch (e) {
+            console.error('\nError: Please specify a valid split amount');
+            process.exit(0);
+        }
+    }
+}
+
+async function mainProcessor(options, processor) {
+    const config = loadSuiConfig(options.env);
+    await processor(config, config.sui, options);
+    saveConfig(config, options.env);
+}
+
+if (require.main === module) {
+    const program = new Command();
+    program.name('manage-coins').description('Merge or split coins for an account');
+
+    addBaseOptions(program);
+
+    program
+        .option('--merge', 'Merge all coins')
+        .option('--split <amount>', 'Split coins')
+        .option('--coin-type <coinType>', 'Coin type to merge/split')
+        .option('--transfer <recipientAddress>', 'Used with --split to transfer the split coins to the recipient address')
```

Comment:
In other scripts, we use the `--action` pattern for the main action, instead of separate cli flags.

Although, I'd like to switch CLI groups going forward, which is the right way to handle CLI cmds (e.g. how `axelard` nests commands). Can you try that out here?
https://github.com/tj/commander.js?tab=readme-ov-file#commands

We can still use the `mainProcessor` pattern, although it can also forward the keypair and client and processCommand can be action specific maybe? broadcast can be a shared helper method.

Reply:
cc @blockchainguyy (similar to the Click refactor in python)

we should refactor other scripts to follow this pattern as well